siva/location.go

```diff
@@ -228,21 +233,32 @@ func (l *Location) Repositories(mode borges.Mode) (borges.RepositoryIterator, er
 
 // Commit persists transactional or write operations performed on the repositories.
 func (l *Location) Commit(mode borges.Mode) error {
-	if !l.lib.transactional || mode != borges.RWMode {
+	if !l.lib.transactional {
+		return borges.ErrNonTransactional.New()
+	}
+
+	if mode != borges.RWMode {
 		return nil
 	}
 
-	defer l.txer.Stop()
+	defer func() {
+		l.cachedFS = nil
```

Comment:
I think that if `l.checkpoint.Reset` fails, we still can use the same `cachedFS`